src/routes/movies.py

```
@@ -1,10 +1,70 @@
-from fastapi import APIRouter, Depends, HTTPException, Query
-from sqlalchemy import select
+from fastapi import APIRouter, Depends, HTTPException, Query, status
+from sqlalchemy import select, func
 from sqlalchemy.ext.asyncio import AsyncSession
+import math
 
 from database import get_db, MovieModel
-
+from schemas.movies import MovieListResponseSchema, MovieDetailResponseSchema
 
 router = APIRouter()
 
-# Write your code here
+
+def get_pagination_params(
+    page: int = Query(1, ge=1),
+    per_page: int = Query(10, ge=1, le=20),
+):
+    return {"page": page, "per_page": per_page}
+
+
+@router.get("/movies/", response_model=MovieListResponseSchema)
+async def read_movies(
+        db: AsyncSession = Depends(get_db),
+        pagination: dict = Depends(get_pagination_params),
+):
+
+    page = pagination["page"]
+    per_page = pagination["per_page"]
+    start = (page - 1) * per_page
+
+    result = await db.execute(
+        select(MovieModel).offset(start).limit(per_page)
+    )
+    movies = result.scalars().all()
+    movie_list = [MovieDetailResponseSchema.model_validate
+                  (movie, from_attributes=True) for movie in movies]
+
+    if not movies:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="No movies found."
+        )
```

This check for an empty `movies` list is causing incorrect behavior. It returns a 404 error if a specific page is empty (e.g., for a page number that is out of bounds), but the requirement is to only return a 404 if there are no movies in the database at all. An out-of-bounds page should return a 200 OK with an empty list of movies. 

Consider fetching the total count of movies first. If the count is zero, then you can raise the 404 exception.